### ISSUE TITLE
Link AdGuard inside Secure Folder to AdGuard outside Secure Folder

### ISSUE THREAD

**Issue by kongacute** [AdguardTeam/AdguardForAndroid#4443]
### Issue Details

AdGuard VPN needs to install both inside and outside of Secure Folder for link AdGuard.

### Proposed solution

Link AdGuard connection to from inside of Secure Folder to AdGuard from outside of Secure Folder via local proxy.

### Alternative solution

N/A

**Comment by kongacute** [comment]
> I second this. It's not a new feature, it's already working that way with the release/beta versions, it's missing in the nightly-versions. 

Did you use a local manual proxy?

**Comment by AlfredSpijker** [comment]
> > I second this. It's not a new feature, it's already working that way with the release/beta versions, it's missing in the nightly-versions. 
> 
> Did you use a local manual proxy?

Yes, I do. Compatability mode is not working from within the Secure folder.

**Comment by kongacute** [comment]
> > > I second this. It's not a new feature, it's already working that way with the release/beta versions, it's missing in the nightly versions. 
> > 
> > Did you use a local manual proxy?
> 
> Yes, I do. The compatibility mode is not working from within the Secure folder.

Ah. Yeah. I feel like it is better to have some proxy server from outside the AdGuard app and route all apps there. But not sure about the cons of this solution.

**Comment by maxikuzmin** [comment]
@AlfredSpijker @kongacute can you tell me what exactly is not working right now, and how do you see Secure folder working with AdGuard?

**Comment by kongacute** [comment]
> @AlfredSpijker @kongacute can you tell me what exactly is not working right now, and how do you see Secure folder working with AdGuard?

The compatibility mode only works with AdGuard where it is installed (outside with outside and inside with inside). Mean I need to turn on 2 AdGuard VPNs both in and outside of Secure Folder if I want to use compatibility mode.

**Comment by maxikuzmin** [comment]
@kongacute As far as I know, this is not possible due to the system because the secure folder is a separate space.
@AlfredSpijker have you applied any additional tools to be compatible with AdGuard VPN from a secure folder? This should also be unavailable in release/beta versions

**Comment by kongacute** [comment]
> @kongacute As far as I know, this is not possible due to the system because the secure folder is a separate space.
> @AlfredSpijker have you applied any additional tools to be compatible with AdGuard VPN from a secure folder? This should also be unavailable in release/beta versions

They still can communicate via proxy server. I set 127.0.0.1:1080 or localhost:1080 (AdGuard VPN set to port 1080 in advance settings).
So I think AdGuard can check by pinging the proxy server. If the proxy server is down then auto-stop routing via that proxy server and vice-versa.

**Comment by AlfredSpijker** [comment]
> @kongacute As far as I know, this is not possible due to the system because the secure folder is a separate space.
> @AlfredSpijker have you applied any additional tools to be compatible with AdGuard VPN from a secure folder? This should also be unavailable in release/beta versions

@maxikuzmin @kongacute 
Sorry for not answering here yet. Will post my config and settings here this weekend.

**Comment by AlfredSpijker** [comment]
@maxikuzmin

I assume you are familiar with the features of Samsung's Secure Folder.

Information:

Device: Samsung S22+ (no root)

Installation in standard environment:
AdGuard v4.0.38 (82)
AdGuard VPN v2.3 Pre-Alpha 3 (20)
Working mode SOCKS5 local proxy (127.0.0.1:1080)

Installation in Secure Folder:
AdGuard v4.0.38 (82)
Working mode SOCKS5 local proxy (127.0.0.1:1080)

Both configurations are set to AdGuard Private DNS (with different keys) for testing purposes.

Comments:
- I don't want to install the newer nightly versions yet because they are still too unstable and this device is partly a production device.
- Compatibility mode in the default environment is not supported by AdGuard with a message that a newer version must be downloaded. That turns out not to be possible.

Both AdGuard in the standard environment and AdGuard in the Secure Folder are therefore connected to AdGuard VPN in the standard environment (i.e. outside the Secure Folder) via the same local proxy.  That works fine.
It is noticeable that when the device is restarted, AdGuard does not automatically connect to the local proxy in the standard environment. I have to enable it manually in Settings > Network. Hopefully this can be fixed.

There was also some hassle with the AdGuard license settings: both installations were initially registered with the same license key (for 4 devices). Regularly 1 of the registrations was deactivated and I had to restore it manually. I thought that was strange, because there is 1 license per device and not 1 per installation.
Apparently you have changed something in the licensing system, because recently both installations with the same license key are accepted in the licensing system, but at the expense of 2 devices, which I think is incorrect.

In short, there is still room for improvement in making the cooperation between the 2 AdGuard installations work by making it possible for both installations (i.e. in the standard environment and in the Secure Folder) to work in the Compatibility (Integraded) mode.

If you want me testing something specific, let me know. As soon as the new nightly version is more stable, I will install it. 
Another factor is that all translations must be available, so that I, as a Dutch translator/proofreader, can also test this.

@kongacute

**Comment by maxikuzmin** [comment]
@AlfredSpijker the only compatibility solution from a secure folder is a local proxy server, you're right.
I checked the proxy problem, my proxy connects automatically after reboot, it does not repeat. I need to take a look at your logs.
What about the license key: it is possible because of the peculiarities of the system and the two spaces, maybe the App ID is recognized as different. You should also take a look at the logs. But I also checked this on version 3.6.11, the license keys are also backed up from the secure folder and the normal environment

Here's what we need you to do:

- Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug);
- Remember the exact time when the issue was reproduced. We will need it to find the corresponding records in the log file;
- Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report);
- Send both files to devteam@adguard.com. Mention the Github issue number and the exact time when the issue was reproduced.
-

**Comment by maxikuzmin** [comment]
@AlfredSpijker any news?

**Comment by maxikuzmin** [comment]
@kongacute as I wrote, "the only compatibility solution from a secure folder is a local proxy server." 
Do you have any difficulty using a proxy server to communicate?

**Comment by kongacute** [comment]
> @kongacute as I wrote, "the only compatibility solution from a secure folder is a local proxy server." 
> Do you have any difficulty using a proxy server to communicate?

Yes. When the proxy was disconnected, AdGuard will still try to connect to that local proxy. Then I got no internet connection.
That is why I sometimes lose internet connection when the proxy won't try to reconnect to their server after being disconnected.

**Comment by maxikuzmin** [comment]
@kongacute sorry, I didn't catch that. Do you still have any questions?

**Comment by kongacute** [comment]
> @kongacute sorry, I didn't catch that. Do you still have any questions?
> 

I mean like this.
AdGuard VPN disconnected → AdGuard app which is not in compatibility mode (as it's not outside of Secure Folder while AdGuard VPN is outside) will still connect to the proxy profile → No internet connection.

**Comment by maxikuzmin** [comment]
@kongacute yes, this is normal behavior, it should be.
There are two ways to solve the problem:
1. You can put the AdGuard VPN in the secure folder as well so that they work together there, but don't forget to change the port in the AdGuard VPN, it is important that it is different from the AdGuard VPN that is outside.
2. Disable the local AdGuard proxy inside the secure folder while AdGuard VPN is disabled from the outside.

**Comment by maxikuzmin** [comment]
@kongacute please clarify, do you have any questions? Were you able to set it up?

**Comment by kongacute** [comment]
> @kongacute please clarify, do you have any questions? Were you able to set it up?

No. I already set it up as you said. It is just about 1 VPN that will not auto-turn off the Proxy if AdGuard VPN is disconnected.

**Comment by maxikuzmin** [comment]
@kongacute then I'll close this issue. If problems or questions arise in the future, be sure to create new issues
